template.go

```diff
@@ -265,8 +265,24 @@ func walkDir(ctx context.Context, cfg *config.Config, dir string, outFileNamer f
 		return nil, fmt.Errorf("ignore matching failed for %s: %w", dir, err)
 	}
 
+	excludeProcessingMatches, err := matcher.Matches(".", &xignore.MatchesOptions{
+		Ignorefile:    gomplateignore,
+		Nested:        true, // allow nested ignorefile
```

Comment:
I think this means that the `.gomplateignore` file will be _also_ considered for `excludeProcessingMatches`, which doesn't make sense.

I'm not sure this will functionally cause a problem (because those files will already be completely ignored, and skipped), but it's liable to cause a bug down the road.

The `xignore` module that I'm using unfortunately doesn't seem to support _not_ specifying an `Ignorefile`. I think I'm going to need to replace that module soon.

For now, how about a comment:

```suggestion
		// TODO: fix or replace xignore module so we can avoid attempting to read the .gomplateignore file for both exclude and excludeProcessing patterns
		Ignorefile:    gomplateignore,
		Nested:        true, // allow nested ignorefile
```